test/select-query-parser/result.test-d.ts

```diff
@@ -70,3 +71,18 @@ type SelectQueryFromTableResult<
   }
   expectType<TypeEqual<typeof result, typeof expected>>(true)
 }
+
+// nested query with selective fields
+{
+  let result: SelectQueryFromTableResult<
+    'users',
+    `msgs:messages(id, ...message_details(created_at, channel!inner(id, slug, owner:users(*))))`
+  >
+  let expected: {
+    msgs: {
+      id: number
+      message_details: SelectQueryError<`Could not embed because more than one relationship was found for 'messages' and '${string}' you need to hint the column with messages!<columnName> ?`>
+    }[]
+  }
+  expectType<TypeEqual<typeof result, typeof expected>>(true)
```

Comment:
Been meaning to ask in the parser rewrite PR, but why use `ts-expect`? This makes it hard to see what exactly went wrong in the CI logs, since the only information it gives is whether the types match or not.

Reply:
I had issue where `expectType` somehow swallowed some of what might be considered types errors (such as additional non expected fields).